core/src/main/java/org/stellar/anchor/util/TransactionHelper.java

```diff
@@ -131,10 +131,15 @@ public static GetTransactionResponse toGetTransactionResponse(
         .kind(PlatformTransactionData.Kind.from(txn.getKind()))
         .status(SepTransactionStatus.from(txn.getStatus()))
         .type(txn.getType())
-        .amountExpected(new Amount(txn.getAmountExpected(), amountExpectedAsset))
-        .amountIn(Amount.create(txn.getAmountIn(), amountInAsset))
-        .amountOut(Amount.create(txn.getAmountOut(), amountOutAsset))
-        .amountFee(Amount.create(txn.getAmountFee(), amountFeeAsset))
+        .amountExpected(
+            (amountExpectedAsset != null)
+                ? new Amount(txn.getAmountExpected(), amountExpectedAsset)
+                : null)
+        .amountIn((amountInAsset != null) ? Amount.create(txn.getAmountIn(), amountInAsset) : null)
```

Comment:
What if asset is not null but amount is null? Do we handle this cases yet?

Reply:
They are actually legit use cases such as: https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/platform/src/test/kotlin/org/stellar/anchor/platform/rpc/DoStellarRefundHandlerTest.kt#L911